src/kixi/nhs/data/transform.clj

```diff
@@ -6,12 +6,16 @@
             [clj-time.core           :as t]
             [clj-time.format         :as tf]))
 
+(defn not-nil? [x] (not (nil? x)))
+
 (defn parse-number
-  "Reads a number from a string. Returns nil if not a number."
+  "Reads a number from a string. Returns nil if not a number
+  or when the value passed is nil."
   [s]
-  (let [parsed (clojure.string/replace s #"," "")]
-    (when (re-find #"^-?\d+\.?\d*$" parsed)
-      (read-string parsed))))
+  (when (not-nil? s)
+    (let [parsed (clojure.string/replace s #"," "")]
+      (when (re-find #"^-?\d+\.?\d*$" parsed)
+        (read-string parsed)))))
```

Comment:
Using `string->int` and `string->float` assumes that I know the kind of numerical data the dataset contains. So if I'm wrong this can happen:
```clojure
=>(string->int "1.678")
   nil
```

If I only use `string->float`, then all my numbers will be floats. This works for further calculations but when writing to CKAN, I convert it to string and I think it's better not to store it in exponential notation.

```clojure
=> (string->float "10000000")
    1.0E7
```

Wdyt?